src/door.rs

```diff
@@ -0,0 +1,291 @@
+        Err(e) => {
+            eprintln!("Couldn't create rundir {0}: {1}", rundir.display(), e);
+            std::process::exit(1);
+        }
+    };
+
+    let door_lockfile_path = rundir.join(format!("{0}.lock", args.door));
+    let door_lockfile = match fs::File::options()
+        .read(true)
+        .write(true)
+        .create(true)
+        .truncate(false)
+        .open(&door_lockfile_path)
+    {
+        Ok(lockfile) => lockfile,
+        Err(e) => {
+            eprintln!(
+                "Couldn't open lockfile {0}: {1}",
+                door_lockfile_path.display(),
+                e
+            );
+            std::process::exit(1);
+        }
+    };
+
+    match door_lockfile.try_lock_shared() {
+        Ok(_) => (),
+        Err(_) => {
+            eprintln!("Sorry, {0} is currently undergoing maintenence.", args.door);
+            std::process::exit(1);
+        }
+    };
+
+    let mut node: i8 = 1;
+    let mut found_node = false;
+
+    while (node <= door.nodes) && !found_node {
+        let node_lockfile_path = rundir.join(format!("{0}.{1}.lock", args.door, node));
+        let node_lockfile = fs::File::options()
+            .read(true)
+            .write(true)
+            .create(true)
+            .truncate(false)
+            .open(&node_lockfile_path);
+
+        if node_lockfile.is_ok() {
+            found_node = match node_lockfile.unwrap().try_lock_exclusive() {
+                Ok(_) => true,
+                Err(_) => false,
+            };
+        }
+
+        if found_node {
+            let node_rundir = rundir.join(format!("{0}.{1}", args.door, node));
+
+            if node_rundir.exists() {
+                match fs::remove_dir_all(&node_rundir) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        eprintln!("Couldn't clean up {0}: {1}", node_rundir.display(), e);
+                        std::process::exit(1)
+                    }
+                };
+            }
+
+            let mkdir = fs::create_dir_all(&node_rundir);
+
+            if mkdir.is_err() {
+                eprintln!(
+                    "Couldn't create node rundir {0}: {1}",
+                    node_rundir.display(),
+                    mkdir.err().unwrap()
+                );
+                std::process::exit(1);
+            }
+
+            let vars = LaunchVars {
+                user: &user,
+                node: node,
+                current_time: Local::now().format("%H:%M").to_string(),
+            };
+
+            let templates = Templates::new();
+
+            match templates.write_dos("door.sys", &node_rundir, &vars) {
+                Ok(_) => (),
+                Err(e) => {
+                    eprintln!("{0}", e);
+                    std::process::exit(1);
+                }
+            };
+
+            let commands = match templates.render_string(&door.launch, &vars) {
+                Ok(commands) => BatchCommands { commands: commands },
```

Comment:
This can be just `BatchCommands { commands }`. Things like this and more should be picked up by `cargo fmt`.